src/components/dashboard/main-content/course-enrollments/data/selectors.js

```diff
@@ -33,20 +33,31 @@ const transformCourseRun = (originalCourseRun) => {
   return courseRun;
 };
 
+export const filterCourseRuns = (courseRuns) => {
+  const courseRunsByStatus = {
+    in_progress: [],
+    upcoming: [],
+    completed: [],
+    savedForLater: [],
+  };
+  if (courseRuns && courseRuns.length > 0) {
+    const transformedCourseRuns = courseRuns.map(transformCourseRun);
+    Object.keys(courseRunsByStatus).forEach((status) => {
+      courseRunsByStatus[status] = transformedCourseRuns.filter((courseRun) => {
+        if (courseRun.courseRunStatus !== 'completed' || (courseRun.courseRunStatus === 'completed' && !courseRun.savedForLater)) {
+          return courseRun.courseRunStatus === status;
+        }
+        if (status === 'savedForLater' && courseRun.savedForLater) {
```

Comment:
Suggest: make this a constant exported string somewhere, and use here as well as in CourseEnrollments.jsx ... ditto for all course run status strings ...